catalog/spatial/registry/registry-ebrim-transformer/src/main/java/org/codice/ddf/registry/converter/RegistryPackageConverter.java

```diff
@@ -414,12 +405,10 @@ private static void parseTopLevel(RegistryObjectType registryObject, MetacardImp
 
         if (registryObject.isSetExternalIdentifier()) {
             List<ExternalIdentifierType> extIds = registryObject.getExternalIdentifier();
-            for (ExternalIdentifierType extId : extIds) {
-                if (extId.getId()
-                        .equals(RegistryConstants.REGISTRY_MCARD_LOCAL_ID)) {
-                    metacard.setId(extId.getValue());
-                }
-            }
+            extIds.stream()
+                    .filter(extId -> extId.getId()
+                            .equals(RegistryConstants.REGISTRY_MCARD_LOCAL_ID))
+                    .forEach(extId -> metacard.setId(extId.getValue()));
```

Comment:
You're a mad man!

I think for-loops are fine, and perfectly readable (sometimes moreso for me). Also, if you only want 1 match and it doesn't matter which, you can add a `break;`.